src/app/core/directives/item-per-page.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { Directive, Input, OnInit } from '@angular/core';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+
+@Directive({
+  selector: '[itemsPerPage]'
+})
+export class ItemsPerPageDirective implements OnInit {
+  private element: MatPaginator;
+  private key: string;
+
+  @Input('itemsPerPage') id: string;
+
+  get pageSize() {
+    const cachedPageSize = sessionStorage.getItem(this.key) || this.element.pageSize;
+    return +cachedPageSize;
```

Comment:
Might be better to use `Number(cachedPageSize)` for greater clarity? And also to standardise with line 19. `Number` behaves in almost exactly the same way as  the unary `+`, and none of the page sizes are one of the funny corner cases anyway.